receiver/zookeeperreceiver/scraper.go

```diff
@@ -119,30 +113,20 @@ func (z *zookeeperMetricsScraper) getResourceMetrics(conn net.Conn) (pdata.Resou
 		return pdata.NewResourceMetricsSlice(), err
 	}
 
-	stats, attributes := z.getMetricsAndAttributes(scanner)
-	metrics := simple.Metrics{
-		Metrics:                    pdata.NewMetrics(),
-		Timestamp:                  time.Now(),
-		InstrumentationLibraryName: "otelcol/zookeeper",
-		MetricFactoriesByName:      metadata.M.FactoriesByName(),
-		ResourceAttributes:         attributes,
-	}
-
-	for _, stat := range stats {
-		// Currently the receiver only deals with one metric type.
-		switch stat.metric.DataType() {
-		case pdata.MetricDataTypeIntGauge:
-			metrics.AddGaugeDataPoint(stat.metric.Name(), stat.val)
-		case pdata.MetricDataTypeIntSum:
-			metrics.AddSumDataPoint(stat.metric.Name(), stat.val)
-		}
+	md := pdata.NewMetrics()
+	z.appendMetrics(scanner, md.ResourceMetrics())
+	mc, _ := md.MetricAndDataPointCount()
+	if mc == 0 {
+		md.ResourceMetrics().Resize(0)
 	}
```

Comment:
Why this this needed?

Reply:
Mostly for tests because we test for empty resource metrics, maybe we should not? Please file a PR or issue if you think this should be removed